### PR TITLE
Scrim and warning dialog

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description
Made a callable scrim and warning dialog.
Reduces code duplication and ensures uniform look throughout the application

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes:
[AB#176816](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/176816)
[AB#176814](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/176814)
[AB#176818](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/176818)
[AB#176821](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/176821)
[AB#176822](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/176822)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/AppBuilder/Types/AppGroupe.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export type AppGroups = Record<Apps, AppGroupe>;
+
+export interface AppGroupe {
+    name: string;
+    icon: string | React.FC;
+}
+
+export enum Apps {
```
Comment (Gustav-Eikaas) [inline src/Core/AppBuilder/Types/AppGroupe.ts:8]: rename to appgroups

Comment (Gustav-Eikaas) [reply src/Core/AppBuilder/Types/AppGroupe.ts:8]: Move to app.ts

File: src/Core/AppBuilder/Types/Manifest.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    appGroups: AppGroups;
+}
+
+type HEXColor = `#${string}`;
+
+export interface AppManifest {
+    title: string;
+    shortName: string;
+    color: HEXColor;
+    groupe: Apps | Apps[];
```
Comment (Gustav-Eikaas) [inline src/Core/AppBuilder/Types/Manifest.ts:15]: string | string[]

File: src/Core/WorkSpace/src/WorkSpaceApi/State.ts [hunk 3]
```diff
@@ -62,16 +61,19 @@ export interface TreeOptions<T> {
...
     gardenKey: keyof T;
     itemKey: keyof T;
-    groupByKeys?: (keyof T)[];
     excludeKeys?: (keyof T)[];
+    groupByKeys?: (keyof T)[];
     customViews?: CustomView<T>;
     options?: Options<T>;
     status?: StatusView<T>;
+    onSelect?: (item: T) => void;
+    disableOnClick?: boolean;
```
Comment (Gustav-Eikaas) [inline src/Core/WorkSpace/src/WorkSpaceApi/State.ts:76]: remove

File: src/apps/Commissioning/index.tsx [hunk 4]
```diff
@@ -79,7 +78,7 @@ const analyticsOptions2: AnalyticsOptions<WP> = {
     },
 };
 
-export function setup(appApi: AppApi): void {
+export function setup(appApi: ClientApi): void {
```
Comment (Gustav-Eikaas) [inline src/apps/Commissioning/index.tsx:81]: Cleanup

File: src/packages/Sidesheet/Components/PopoutSidesheet.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    position: absolute;
+    top: 0px;
+    right: 0px;
+    width: auto;
+    height: 100%;
+    background: white;
+    border-left: 2px solid ${tokens.colors.ui.background__medium.rgba};
+`;
+
+// borderleft: 2px solid ${tokens.colors.ui.background__medium.rgba};
```
Comment (Gustav-Eikaas) [inline src/packages/Sidesheet/Components/PopoutSidesheet.tsx:32]: remove

File: src/packages/Sidesheet/context/sidesheetContext.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { Atom } from '@dbeining/react-atom';
+import { createGlobalScrimState } from '../State/actions';
+import { SidesheetState } from '../State/sidesheetState';
+
+export const SidesheetCoreContext = createGlobalScrimState({});
```
Comment (Gustav-Eikaas) [inline src/packages/Sidesheet/context/sidesheetContext.ts:5]: rename

**Review by Noggling — APPROVED** [review]

**Pull request merged**